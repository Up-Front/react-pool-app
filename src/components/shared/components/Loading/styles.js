import styled, { keyframes } from 'styled-components';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const Rotate = styled.div`
  display: inline-block;
  animation: ${rotate360} 2s linear infinite;
  padding: 2rem 1rem;
  font-size: 1.2rem;
`;
