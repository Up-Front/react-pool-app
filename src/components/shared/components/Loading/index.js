import React from 'react';
import { Rotate } from './styles';

const Loading = () => (
  <Rotate>
    <span role="img" aria-label="left hand">
      👈
    </span>
    <span role="img" aria-label="8-ball">
      🎱
    </span>
    <span role="img" aria-label="right hand">
      👉
    </span>
  </Rotate>
);

export default Loading;
