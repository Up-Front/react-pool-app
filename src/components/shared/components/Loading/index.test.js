import React from 'react';
import { shallow } from 'enzyme';
import Constants from './../../../shared/constants';
import Loading from './';

describe('Loading Component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
