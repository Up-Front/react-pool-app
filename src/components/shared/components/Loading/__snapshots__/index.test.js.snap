// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span
          aria-label="left hand"
          role="img"
>
          ðŸ‘ˆ
</span>,
        <span
          aria-label="8-ball"
          role="img"
>
          ðŸŽ±
</span>,
        <span
          aria-label="right hand"
          role="img"
>
          ðŸ‘‰
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "left hand",
          "children": "ðŸ‘ˆ",
          "role": "img",
        },
        "ref": null,
        "rendered": "ðŸ‘ˆ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "8-ball",
          "children": "ðŸŽ±",
          "role": "img",
        },
        "ref": null,
        "rendered": "ðŸŽ±",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "right hand",
          "children": "ðŸ‘‰",
          "role": "img",
        },
        "ref": null,
        "rendered": "ðŸ‘‰",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span
            aria-label="left hand"
            role="img"
>
            ðŸ‘ˆ
</span>,
          <span
            aria-label="8-ball"
            role="img"
>
            ðŸŽ±
</span>,
          <span
            aria-label="right hand"
            role="img"
>
            ðŸ‘‰
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "left hand",
            "children": "ðŸ‘ˆ",
            "role": "img",
          },
          "ref": null,
          "rendered": "ðŸ‘ˆ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "8-ball",
            "children": "ðŸŽ±",
            "role": "img",
          },
          "ref": null,
          "rendered": "ðŸŽ±",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "right hand",
            "children": "ðŸ‘‰",
            "role": "img",
          },
          "ref": null,
          "rendered": "ðŸ‘‰",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
